### PR TITLE
Feature/devsu 2412 optionally order by evidencelevel in printtable

### PR DESCRIPTION
**Pull request by elewis2**
Adds option to sort PrintTables by interior column (not one of the collapsible columns). Applied to therapeuticTargets tables in this pr.

### CODE REVIEW

File: app/components/PrintTable/index.tsx [hunk 1]
```diff
@@ -84,13 +88,62 @@ function PrintTable({
...
+            acc[rowkey] = {};
+          }
+          Object.keys(row).forEach((key) => {
+            if (outerRowOrderByInternalCol.includes(key)) {
+              if (!acc[rowkey][key]) {
+                acc[rowkey][key] = [];
+              }
+              acc[rowkey][key].push(row[key]);
+            }
+          })
```
Comment (bnguyen-bcgsc) [inline app/components/PrintTable/index.tsx:106]: missing semicolon here

File: app/components/PrintTable/index.tsx [hunk 2]
```diff
@@ -84,13 +88,62 @@ function PrintTable({
...
+        if (outerRowOrderByInternalCol) {
+          // order by 'outerRowOrderByInternalCol' if possible, then by outer row key if necessary
+          for (let i = 0; i < outerRowOrderByInternalCol.length; i++) {
+            const rankA = outerRowsRank[aKey][outerRowOrderByInternalCol[i]];
+            const rankB = outerRowsRank[bKey][outerRowOrderByInternalCol[i]];
+            if (!(rankA === rankB)) {
+              return rankA > rankB ? 1 : -1;
+            }
+          }
         return aKey > bKey ? 1 : -1;
```
Comment (bnguyen-bcgsc) [inline app/components/PrintTable/index.tsx:145]: missing 2 indents

**Review by bnguyen-bcgsc — CHANGES_REQUESTED** [review]
srry just some small linting changes, otherwise everything is working on my end

File: app/components/PrintTable/index.tsx [hunk 3]
```diff
@@ -84,13 +88,62 @@ function PrintTable({
...
+        }, {});
+
+        Object.keys(outerRowsRank).forEach((key) => {
+          Object.keys(outerRowsRank[key]).forEach((subkey) => {
+            outerRowsRank[key][subkey] = JSON.stringify(outerRowsRank[key][subkey].sort()); // Sort and stringify
+          });
+        });
+      }
 
       (collapseableCols?.length > 0 ? [...data].sort((aRow, bRow) => {
```
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/PrintTable/index.tsx:117]: return a 0 at line 147?

[the remark this answers is not in the document]
Comment (bnguyen-bcgsc) [reply app/components/PrintTable/index.tsx:117]: that's what I was thinking too

**Review by bnguyen-bcgsc — APPROVED** [review]

**Pull request merged**